File: src/main/java/info/matsumana/tsujun/model/ksql/KsqlResponseColumns.java
package info.matsumana.tsujun.model.ksql;

import java.util.Arrays;

public class KsqlResponseColumns {

    private Object[] columns;

    public Object[] getColumns() {
        return columns;
    }

    public void setColumns(Object[] columns) {
        this.columns = columns;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) { return true; }
        if (o == null || getClass() != o.getClass()) { return false; }
        KsqlResponseColumns that = (KsqlResponseColumns) o;
        return Arrays.equals(columns, that.columns);
    }

    @Override
    public int hashCode() {
        return Arrays.hashCode(columns);
    }

    @Override
    public String toString() {
        return "KsqlResponseColumns{" +
               "columns=" + Arrays.toString(columns) +
               '}';
    }
}

File: src/main/kotlin/info/matsumana/tsujun/model/ksql/KsqlResponseColumns.kt
package info.matsumana.tsujun.model.ksql

import java.util.*

data class KsqlResponseColumns(val columns: Array<Any>) {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as KsqlResponseColumns

        if (!Arrays.equals(columns, other.columns)) return false

        return true
    }

    override fun hashCode(): Int {
        return Arrays.hashCode(columns)
    }
}
